src/cbuild/hooks/do_pkg/000_collect_rdeps.py

```diff
@@ -0,0 +1,71 @@
+from cbuild.core import logger
+from cbuild.apk import cli
+
+
+def collect_deps(pkg):
+    # dependencies of any sort
+    deps = []
+
+    # bootstrap packages are not installable ootb
+    if pkg.pkgname.endswith("-bootstrap") and pkg.build_style != "meta":
+        deps += ["bootstrap:" + pkg.pkgname.removesuffix("-bootstrap")]
+
+    # explicit package depends
+    for c in pkg.depends:
+        ploc = c.find("!")
+        if ploc > 0:
+            deps.append(c[0:ploc].removeprefix("virtual:"))
+        else:
+            deps.append(c.removeprefix("virtual:"))
+
+    # sort before adding more
+    deps.sort()
+
+    # shlib requires
+    if hasattr(pkg, "so_requires"):
+        deps += map(lambda v: f"so:{v}", sorted(pkg.so_requires))
+
+    # .pc file requires
+    if hasattr(pkg, "pc_requires"):
+        deps += map(lambda v: f"pc:{v}", sorted(pkg.pc_requires))
+
+    pkg.all_rdeps = deps
+
+
+def print_deps_diff(pkg):
+    subp = cli.call_chroot(
+        "info", ["--depends", "--", pkg.pkgname], pkg, capture_output=True
+    )
+
+    log = logger.get()
+
+    if subp.returncode != 0:
+        log.warn(f'Could not get original dependencies of "{pkg.pkgname}"')
+        return
+
+    log.out("Dependency comparison between original and newly-built package:")
+
+    # first line is e.g. "foo-1.0.0-r0 depends on:", so skip it
+    orig_rdeps = subp.stdout.decode().strip().split("\n\n")[0].split("\n")[1:]
+
+    list1s = set(orig_rdeps)
+    list2s = set(pkg.all_rdeps)
+
+    lists = set(list1s)
+    lists.update(list2s)
+
+    for v in sorted(lists):
+        if v in list1s:
+            if v not in list2s:
+                log.out_red(f"-{v}")
+                continue
+        else:
+            if v in list2s:
+                log.out_green(f"+{v}")
+                continue
+        log.out(f" {v}")
```

Comment:
the fat arrows added by the logger before each line are kinda annoying here, idk